projects/motor_controller_interface/README.md

```diff
@@ -0,0 +1,107 @@
+
+In addition, we need to perform some bookkeeping to ensure that we are sending 
+the correct values. This can be done in the output functions.
+
+A `CanInterval` is used to periodically send the messages to each of the motor 
+controllers. Since the `CanInterval` object is intended to be used for a 
+persistent periodic message, we directly manipulate the stored object before 
+each CAN transmission.
+
+Each Motor Controller has a unique ID. Moreover, each *logical* Motor 
+Controller Interface *also* has a unique ID. This allows Motor Controllers to 
+distinguish messages sent to each, and allows us to distinguish telemetry 
+messages from each (since CAN is a message-based protocol).
+
+```dot
+digraph {
+  interface_left;
+  interface_right;
+  motor_controller_left;
+  motor_controller_right;
+
+  interface_left -> motor_controller_left [label="Drive commands\n(left Driver Controls Base Address)"];
+  interface_right -> motor_controller_right [label="Drive commands\n(right Driver Controls Base Address)"];
+  motor_controller_left -> interface [label="Telemetry messages\n(left Motor Controller base address)"];
+  motor_controller_right -> interface [label="Telemetry messages\n(right Motor Controller base address)"];
+  subgraph cluster {
+    {rank=same interface_left interface interface_right};
+
+    interface_left -> interface -> interface_right [arrowhead=none];
+  }
+}
+```
+
+![](https://chart.googleapis.com/chart?cht=gv&chl=digraph%20{interface_left;interface_right;motor_controller_left;motor_controller_right;interface_left%20-%3E%20motor_controller_left%20[label=%22Drive%20commands\n(left%20Driver%20Controls%20Base%20Address)%22];interface_right%20-%3E%20motor_controller_right%20[label=%22Drive%20commands\n(right%20Driver%20Controls%20Base%20Address)%22];motor_controller_left%20-%3E%20interface%20[label=%22Telemetry%20messages\n(left%20Motor%20Controller%20base%20address)%22];motor_controller_right%20-%3E%20interface%20[label=%22Telemetry%20messages\n(right%20Motor%20Controller%20base%20address)%22];subgraph%20cluster%20{{rank=same%20interface_left%20interface%20interface_right};interface_left%20-%3E%20interface%20-%3E%20interface_right%20[arrowhead=none];}})
```

Comment:
TBH I'm totally abusing the deprecated Google Charts API to render the GraphViz graphs as images :laughing: .

I thought it'd be nice to render the diagrams, but I can remove them if anyone doesn't like that.